test/integration/acceptance/fake_peer_example_test.cpp

```diff
@@ -314,3 +314,80 @@ TEST_F(FakePeerExampleFixture,
   EXPECT_TRUE(got_message.load())
       << "Reached timeout waiting for an ordering message.";
 }
+
+/**
+ * Check that after receiving a valid command the ITF peer provides a proposal
+ * containing it.
+ *
+ * \attention this code is nothing more but an example of Fake Peer usage
+ *
+ * @given a network of two iroha peers
+ * @when a valid command is sent to one
+ * @then it must either (on demand) provide a proposal containing this command,
+ * or request it from the other peer
+ */
+TEST_F(FakePeerExampleFixture,
+       OnDemandOrderingProposalAfterValidCommandReceived) {
+  static constexpr std::chrono::seconds kProposalWaitingTime(1);
+
+  /* A custom behaviour that requests a proposal for the round it got vote for,
+   * and if gets one, checks that the proposal contains the given tx hash.
+   */
+  struct CustomBehaviour : public fake_peer::HonestBehaviour {
```

Comment:
CustomBehaviour is not a very descriptive name.

Reply:
It is local to this test function, and described at declaration, so I consider a longer descriptive name unneeded here.